src/otk/command.py

```diff
@@ -19,144 +15,169 @@
-    "-i",
-    "--identifier",
-    help="An identifier to include in all log records generated by `otk -j`. Can only be used together with `-j`.",
-)
-@click.option(
-    "-w",
-    "--warn",
-    type=click.Choice(["duplicate-definition"]),
-    multiple=True,
-    help="Enable warnings, can be passed multiple times.",
-)
-@click.pass_context
-def root(
-    ctx: click.Context, verbose: int, json: bool, identifier: str, warn: list[str]
-) -> None:
-    """`otk` is the omnifest toolkit. A program to work with omnifest inputs
-    and translate them into the native formats for image build tooling."""
+def root():
+    argv = sys.argv[1:]
 
-    ctx.ensure_object(dict)
+    parser = parser_create()
+    arguments = parser.parse_args(argv)
 
+    # turn on logging as *soon* as possible so it can be used early
     logging.basicConfig(
-        level=logging.WARNING - (10 * verbose),
+        level=logging.WARNING - (10 * arguments.verbose),
         handlers=[
             (
-                JSONSequenceHandler(identifier, stream=sys.stderr)
-                if json
+                JSONSequenceHandler(arguments.identifier, stream=sys.stderr)
+                if arguments.json
                 else logging.StreamHandler()
-            )
+            ),
         ],
     )
 
-    # This context is passed along to all other subcommands.
-    ctx.obj["context"] = {
-        "duplicate_definitions_warning": "duplicate-definition" in warn,
-    }
+    # some arguments need to be handled before continuing with
+    # subcommands
+    if arguments.identifier and not arguments.json:
+        parser.print_help()
+        parser.exit()
+
+    if arguments.command == "compile":
+        return compile(parser, arguments)
+    else:
+        raise RuntimeError("Unknown subcommand")
+
 
-    # We do this check *after* setting up the handlers so the error is formatted
-    if identifier and not json:
-        log.error("cannot use `-i` without also using `-j`")
-        sys.exit(1)
-
-
-@root.command()
-@click.argument("input", type=click.Path(exists=True))
-@click.argument("output", type=click.Path(), required=False)
-@click.option(
-    "--external/--no-external",
-    default=True,
-    help="Process external directives.",
-)
-@click.option("-t", "--target", "requested_target", required=False)
-@click.pass_context
 def compile(
-    ctx: click.Context,
-    input: str,
-    output: Optional[str],
-    external: bool,
-    requested_target: str,
+    parser: argparse.ArgumentParser, arguments: argparse.Namespace
 ) -> None:
-    """Compile a given omnifest into its targets."""
+    src = pathlib.Path(
+        "/dev/stdin" if arguments.input is None else arguments.input
+    )
+    dst = pathlib.Path(
+        "/dev/stdout" if arguments.output is None else arguments.output
+    )
 
-    log.info("Compiling the input file %r to %r", input, output or "STDOUT")
+    if not src.exists():
+        log.fatal("INPUT path %r does not exist", str(src))
+        return 1
```

Comment:
Function's type annotation should say it returns an `int` now.

Reply:
Added a commit to fix the signature.